aws/resource_aws_cloudwatch_event_connection.go

```diff
@@ -0,0 +1,785 @@
+			},
+			"auth_parameters": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"api_key": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							ExactlyOneOf: []string{
+								"auth_parameters.0.api_key",
+								"auth_parameters.0.basic",
+								"auth_parameters.0.oauth",
+							},
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"key": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"value": {
+										Type:      schema.TypeString,
+										Required:  true,
+										Sensitive: true,
+									},
+								},
+							},
+						},
+						"basic": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							ExactlyOneOf: []string{
+								"auth_parameters.0.api_key",
+								"auth_parameters.0.basic",
+								"auth_parameters.0.oauth",
+							},
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"username": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"password": {
+										Type:      schema.TypeString,
+										Required:  true,
+										Sensitive: true,
+									},
+								},
+							},
+						},
+						"oauth": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							ExactlyOneOf: []string{
+								"auth_parameters.0.api_key",
+								"auth_parameters.0.basic",
+								"auth_parameters.0.oauth",
+							},
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"authorization_endpoint": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"http_method": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validation.StringInSlice(events.ConnectionOAuthHttpMethod_Values(), true),
+									},
+									"oauth_http_parameters": {
+										Type:     schema.TypeList,
+										Required: true,
+										MaxItems: 1,
+										Elem:     connectionHttpParameters,
+									},
+									"client_parameters": {
+										Type:     schema.TypeList,
+										Optional: true,
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"client_id": {
+													Type:     schema.TypeString,
+													Required: true,
+												},
+												"client_secret": {
+													Type:      schema.TypeString,
+													Required:  true,
+													Sensitive: true,
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+						"invocation_http_parameters": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem:     connectionHttpParameters,
+						},
+					},
+				},
+			},
+			"arn": {
```

Comment:
could you also add `secret_arn` as that will be returned if any secret value is passed for params?

Reply:
Will you only get one `secret_arn` back or will it be one per param?